src/sdk/client.js

```diff
@@ -183,7 +244,110 @@ class Client {
       return Promise.reject(e);
     }
   }
+
+  /**
+   * Creates or updates the remote build object.
+   *
+   * @param {object} params Parameters for the build
+   * @returns {Promise<object>} The parsed server response
+   */
+  createOrUpdateBuild(params) {
+    try {
+      const buildParams = Object.assign({}, params);
+      const base = getHookBase();
+      if (!buildParams.build) {
+        throw new Error('Missing build ID');
+      }
+
+      buildParams.label = buildParams.buildLabel;
+
+      const url = new URL(
+        `builds/${buildParams.build}`,
+        sanitizeURL(base)
+      ).toString();
+
+      // Remove non-Zeus keys
+      Object.keys(buildParams).forEach(key => {
+        if (!BUILD_ATTRIBUTES.includes(key)) {
+          delete buildParams[key];
+        }
+      });
+
+      return this.requestJson(url, {
+        method: 'POST',
+        body: buildParams,
+      });
+    } catch (e) {
+      return Promise.reject(e);
+    }
+  }
+
+  /**
+   * Creates or updates the remote job object.
+   *
+   * @param {object} params Parameters for the job
+   * @returns {Promise<object>} The parsed server response
+   */
+  createOrUpdateJob(params) {
+    try {
+      const base = getHookBase();
+      if (!params.job) {
+        throw new Error('Missing job ID');
+      } else if (!params.build) {
+        throw new Error('Missing build ID');
+      }
+
+      const addBuildPromise = this.createOrUpdateBuild(params);
```

Comment:
I'd not do this here and instead require the caller to do this (i.e. the command). What do you think?